Source/Microsoft.Teams.Apps.FAQPlusPlus.Common/Providers/IQnaServiceProvider.cs

```diff
@@ -44,8 +44,9 @@ public interface IQnaServiceProvider
         /// </summary>
         /// <param name="question">Question text.</param>
         /// <param name="isTestKnowledgeBase">Prod or test.</param>
+        /// <param name="payload">The response card payload that contains the previous question when a follow up is queried.</param>
         /// <returns>QnaSearchResult object as response.</returns>
-        Task<QnASearchResultList> GenerateAnswerAsync(string question, bool isTestKnowledgeBase);
+        Task<QnASearchResultList> GenerateAnswerAsync(string question, bool isTestKnowledgeBase, ResponseCardPayload payload = null);
```

Comment:
Can we pass PreviousQnAId and PreviousUserQuery as parameters here? The "ResponseCardPayload" in the signature breaks the abstraction because the concept of a response card payload doesn't quite fit with the concept of an  `IQnAServiceProvider`.

Reply:
Changed to capture only required parameters.